api/stream.go

```diff
@@ -124,23 +117,63 @@ func (s *StreamAPI) newSubscription(
 
 // NewHeads send a notification each time a new block is appended to the chain.
 func (s *StreamAPI) NewHeads(ctx context.Context) (*rpc.Subscription, error) {
-	return s.newSubscription(ctx, s.blocksBroadcaster, func(blockData any) (any, error) {
-		block, ok := blockData.(*types.Block)
-		if !ok {
-			return nil, fmt.Errorf("received data of incorrect type, it should be of type *types.Block")
-		}
+	return s.newSubscription(
+		ctx,
+		s.blocksBroadcaster,
+		func(ctx context.Context, height uint64) (interface{}, error) {
+			block, err := s.blocks.GetByHeight(height)
+			if err != nil {
+				if errors.Is(err, storageErrs.ErrNotFound) { // make sure to wrap in not found error as the streamer expects it
+					return nil, errors.Join(flowgoStorage.ErrNotFound, err)
```

Comment:
the latest version added support for a new sentinel to signal the data is not available yet:
https://github.com/onflow/flow-go/blob/master/engine/access/subscription/streamer.go#L18

let's switch to that when possible